corehq/motech/dhis2/entities_helpers.py

```diff
@@ -209,17 +237,113 @@ def register_tracked_entity_instance(requests, case_trigger_info, case_config):
     summaries = response.json()["response"]["importSummaries"]
     if len(summaries) != 1:
         raise Dhis2Exception(_(
-            f'{len(summaries)} tracked entity instances registered from '
-            f'{case_trigger_info}.'
-        ))
+            '{n} tracked entity instances were registered from case {case}.'
+        ).format(n=len(summaries), case=case_trigger_info))
     if case_config["tei_id"] and "case_property" in case_config["tei_id"]:
         case_property = case_config["tei_id"]["case_property"]
         tei_id = summaries[0]["reference"]
         case_updates[case_property] = tei_id
+        # Store `tei_id` in `case_trigger_info` because we might need it
+        # when we create relationships.
+        case_trigger_info.extra_fields[case_property] = tei_id
     if case_updates:
         save_case_updates(requests.domain_name, case_trigger_info.case_id, case_updates)
 
 
+def create_relationships(
+    requests,
+    subcase_trigger_info,
+    subcase_config,
+    dhis2_entity_config,
+):
+    """
+    Creates two relationships in DHIS2; one corresponding to a
+    subcase-to-supercase relationship, and the other corresponding to a
+    supercase-to-subcase relationship.
```

Comment:
@kaapstorm I'm curious what the impact is when only one of the two relationships gets created (for instance, if `subcase_to_supercase_dhis2_id` is not provided, but `supercase_to_subcase_dhis2_id` is)?

Reply:
It is possible to configure only one of the two directions in DHIS2, and so I think this is a use case that we should be able to support (even though I'm not sure why anyone would want LESS information when MORE information is trivial to offer).